pkg/ingester/user_tsdb.go

```diff
@@ -85,8 +86,9 @@ type userTSDB struct {
 	ingestedRuleSamples *util_math.EwmaRate
 
 	// Cached shipped blocks.
-	shippedBlocksMtx sync.Mutex
-	shippedBlocks    map[ulid.ULID]struct{}
+	shippedBlocksMtx  sync.Mutex
+	shippedBlocks     map[ulid.ULID]time.Time
+	blockMinRetention time.Duration
```

Comment:
[nit] I'd separate this field from previous two, as now it looks like it's protected by `shippedBlocksMtx`, which is not true. (We can eg. move it before `shippedBlocks*` fields).